src/cloud_provider/service.rs

```diff
@@ -1161,6 +1168,98 @@ pub fn do_stateless_service_cleanup(
     Ok(())
 }
 
+/// This function call (start|pause|delete)_in_progress function every 10 seconds when a
+/// long blocking task is running.
+pub fn send_progress_on_long_task<S, R, F>(service: &S, action: Action, long_task: F) -> R
+where
+    S: Service + Listen,
+    F: Fn() -> R,
+{
+    let waiting_message = match action {
+        Action::Create => Some(format!(
+            "{} '{}' deployment is in progress...",
+            service.service_type().name(),
+            service.name_with_id()
+        )),
+        Action::Pause => Some(format!(
+            "{} '{}' pause is in progress...",
+            service.service_type().name(),
+            service.name_with_id()
+        )),
+        Action::Delete => Some(format!(
+            "{} '{}' deletion is in progress...",
+            service.service_type().name(),
+            service.name_with_id()
+        )),
+        Action::Nothing => None,
+    };
+
+    send_progress_on_long_task_with_message(service, waiting_message, action, long_task)
+}
+
+/// This function call (start|pause|delete)_in_progress function every 10 seconds when a
+/// long blocking task is running.
+pub fn send_progress_on_long_task_with_message<S, M, R, F>(
+    service: &S,
+    waiting_message: Option<M>,
+    action: Action,
+    long_task: F,
+) -> R
+where
+    S: Service + Listen,
+    M: Into<String>,
+    F: Fn() -> R,
+{
+    let listeners = std::clone::Clone::clone(service.listeners());
+
+    let progress_info = ProgressInfo::new(
+        service.progress_scope(),
+        Info,
+        match waiting_message {
+            Some(message) => Some(message.into()),
+            _ => None,
```

Comment:
Waiting message should not be an option, otherwise, I don't get the interest in it

Reply:
We don't necessarily show the progress to the user, this function can be used for anything, and the progress can be used to know if the task is really running or not.